torch/serialization.py

```diff
@@ -318,6 +318,16 @@ def restore_location(storage, location):
     elif isinstance(map_location, _string_classes):
         def restore_location(storage, location):
             return default_restore_location(storage, map_location)
+    elif isinstance(map_location, torch.device):
```

Comment:
this should merge with the above case, because `device` arguments should also accept strings and device ordinals (at least for now).

Reply:
Do you want me to change default_restore_location to take torch.devices as well as strings?

Reply:
Since we are letting it take in `device`, then it should take in all things that can be passed as a `device`, which includes `torch.device`, strings, and device ordinals.

Are there more @gchanan ?

Reply:
I think we should accept `torch.device` and `strings`. Device ordinals should be considered legacy thing, so let's avoid adding them.